### ISSUE TITLE
Unable to enable type resolution

### ISSUE THREAD

**Issue by bbaldino** [Ozsie/detekt-maven-plugin#60]
I'll apologize because I'm not sure this is a valid issue, but I've been unable to find any way to make this work.  Enabling type resolution requires setting `jvmTarget` and `classPath`, but I'm unable to find a way to specify a classpath in maven which will include the kotlin stdlib jars, and so detekt fails to parse the kotlin code correctly (and therefore can't warn on failures).

I have a project which illustrates the issue [here](https://github.com/bbaldino/detekt-repro): you can check it out and run `mvn detekt:check -Ddetekt.config=detekt.yml`.  It will catch the use of the forbidden Java method call, but not the Kotlin one (it will also complain about not accessing `kotlin.Int`).

If I copy `kotlin-stdlib-1.3.72.jar` into the project directory and change [this line](https://github.com/bbaldino/detekt-repro/blob/master/pom.xml#L63) to:

```
<classPath>${java.class.path};${project.basedir}/kotlin-stdlib-1.3.72.jar</classPath>
```

Then it works fine, but I can't find any way to dynamically refer to a classpath in maven which will include the kotlin libs, is it possible?

**Comment by Ozsie** [comment]
Maybe i'm missing something, but running `mvn detekt:check -Ddetekt.config=detekt.yml` fails
`
[INFO] Scanning for projects...
Downloading from central: http://jcenter.bintray.com/org/jetbrains/kotlin/kotlin-maven-plugin/1.3.72/kotlin-maven-plugin-1.3.72.pom
[WARNING] Failed to retrieve plugin descriptor for org.jetbrains.kotlin:kotlin-maven-plugin:1.3.72: Plugin org.jetbrains.kotlin:kotlin-maven-plugin:1.3.72 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.jetbrains.kotlin:kotlin-maven-plugin:jar:1.3.72
[INFO]
[INFO] ------------------------< groupId:detekt-repro >------------------------
[INFO] Building detekt-repro 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] Failed to retrieve plugin descriptor for org.jetbrains.kotlin:kotlin-maven-plugin:1.3.72: Plugin org.jetbrains.kotlin:kotlin-maven-plugin:1.3.72 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.jetbrains.kotlin:kotlin-maven-plugin:jar:1.3.72
Downloading from central: http://jcenter.bintray.com/org/jetbrains/kotlin/kotlin-stdlib-jdk8/1.3.72/kotlin-stdlib-jdk8-1.3.72.pom
Downloading from central: http://jcenter.bintray.com/org/jetbrains/kotlin/kotlin-test/1.3.72/kotlin-test-1.3.72.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.781 s
[INFO] Finished at: 2020-07-19T11:56:25+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project detekt-repro: Could not resolve dependencies for project groupId:detekt-repro:jar:1.0-SNAPSHOT: Failed to collect dependencies at org.jetbrains.kotlin:kotlin-stdlib-jdk8:jar:1.3.72: Failed to read artifact descriptor for org.jetbrains.kotlin:kotlin-stdlib-jdk8:jar:1.3.72: Could not transfer artifact org.jetbrains.kotlin:kotlin-stdlib-jdk8:pom:1.3.72 from/to central (http://jcenter.bintray.com): Access denied to: http://jcenter.bintray.com/org/jetbrains/kotlin/kotlin-stdlib-jdk8/1.3.72/kotlin-stdlib-jdk8-1.3.72.pom -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
`

**Comment by bbaldino** [comment]
Hm, I don't see that error (even after clearing out my maven cache).  Maybe it was a transient issue with the repo?  What maven version are you on?  I'm on 3.6.0.

**Comment by Ozsie** [comment]
The error was caused by my borked settings.xml. Having fixed that I am able to reproduce the error, however I'm not sure I'm the right person to fix it.

If I run detekt cli directly, I get the same issue so I don't think the detekt-maven-plugin does anything wrong.
```
detekt -c detekt.yml --jvm-target 1.8 -cp "~/.sdkman/candidates/maven/current/bin:~/.sdkman/candidates/java/current/bin:~/.sdkman/candidates/gradle/current/bin:~/.nvm/versions/node/v12.4.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
warning: classpath entry points to a non-existent location: ~/.sdkman/candidates/maven/current/bin:~/.sdkman/candidates/java/current/bin:~/.sdkman/candidates/gradle/current/bin:~/.nvm/versions/node/v12.4.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
/Main.kt:3:19: error: cannot access built-in declaration 'kotlin.Int'. Ensure that you have a dependency on the Kotlin standard library
    fun answer(): Int = 42
                  ^
/Main.kt:7:5: error: unresolved reference: println
    println("hello world")
    ^
```

I doubt anything is wrong with detekt since we tell it what the classpah is, in this case the JDK, and Kotlin is not part of the JDK. When you add the kotlin std jar to the path it works, so the question is how to add that library to the class path in a nice way.

If you change your classPath configuration to this, it will work without copying the jar to your project.
`<classPath>${java.class.path};${settings.localRepository}/org/jetbrains/kotlin/kotlin-stdlib/${kotlin.version}/kotlin-stdlib-${kotlin.version}.jar</classPath>`

It does look a bit clunky, but I don't think there is any other way to do it while maintaining portability without copying the jar to the project.

I guess you could shorten it a bit by storing `settings.localRepository` and `kotlin.version` in variables with shorter names.

**Comment by bbaldino** [comment]
Thanks @Ozsie!  That's not the worst workaround, actually.  I did find some maven plugins which said they'd allow referring to a dependency jar (which I assume would be some sort of shortcut for what you suggested) but couldn't get any to work.  I also tried asking about how gradle derives this path (since it seems to work there) but didn't get any response.  In the meantime I think I can work with this.  Thanks!
